tools/converter.py

```diff
@@ -0,0 +1,169 @@
+hex_format = '%02x%02x%02x'
+
+red_comp = 'Red Component'
+green_comp = 'Green Component'
+blue_comp = 'Blue Component'
+
+
+class Converter(object):
+    def __init__(
+            self,
+            schemes: list,
+            templates: list,
+            loader: Environment,
+            bar: Progress,
+            path_to_iterm_schemes: str,
+            path_to_xrdb: str,
+            output_dir: str
+    ):
+        self.bar = bar
+        self.loader = loader
+        self.iterm_dir = path_to_iterm_schemes
+        self.xrdb_dir = path_to_xrdb
+        self.out_dir = output_dir
+
+        self.templates = self.get_all_templates(templates)
+
+        self.schemes = schemes
+        if self.schemes is None:
+            self.schemes = self.get_all_schemes()
+
+    def get_all_schemes(self):
+        schemes = []
+
+        for name in os.listdir(self.iterm_dir):
+            schemes.append(iterm_re.match(name).group(1))
+
+        return schemes
+
+    def get_all_templates(self, templates_arg):
+        templates = {}
+
+        for template in self.loader.list_templates():
+            name, ext = os.path.splitext(template)
+
+            if templates_arg is None or name in templates_arg:
+                templates.update({name: template})
+
+        return templates
+
+    def parse_scheme(self, scheme):
+        colors_dict = {}
+
+        iterm_path = self.iterm_dir + scheme + iterm_ext
+        xrdb_path = self.xrdb_dir + scheme + xrdb_ext
+
+        if not os.path.isfile(iterm_path):
+            logging.error('Scheme ' + iterm_path + ' doesn\'t exist')
+            sys.exit(1)
+
+        with open(iterm_path, 'rb') as f:
+            plist = plistlib.load(f)
+            guint16_pallete = []
+
+            for color_name in plist:
+                color_components = plist[color_name]
+                color_hex, rgb = self.calculate_color(color_components)
+                guint16 = self.calculcate_color_components_guint16(color_components)
+                colors_dict[color_name.replace(' ', '_')] = {
+                    'hex': color_hex,
+                    'rgb': ','.join(map(lambda x: str(x), rgb)),
+                    'guint16': guint16
+                }
+
+            # we need second loop because guint16 palette must have colors in sorted order
+            for color_index in range(16):
+                key = 'Ansi %d Color' % color_index
+                guint16_pallete += self.calculcate_color_components_guint16(plist[key])
+            
+            colors_dict['Guint16_Palette'] = '{%s}' % ', '.join(guint16_pallete)
+            colors_dict['Dark_Theme'] = self.detect_dark_theme(int(colors_dict['Background_Color']['hex'], 16))
+
+        f.close()
+
+        # xrdb files were used to generate themes, now it is not needed
+        # but in case someone uses them for other purposes, these files continue to be saved
+        if not os.path.isfile(xrdb_path):
+            self.generate_xrdb_file(xrdb_path, colors_dict)
+
```

Comment:
I think that's probably a good move, but is there any reason this couldn't also be moved to a template? As written, couldn't calling `python3 gen.py -t foo` generate themes for `foo` and `xrdb` formats?